Bottom: c43ee60463f8965f259bff46add1c303340fefed
Top:    6bfd663f1747b7d4d03f101e5d6e42e1c1652510
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-24 15:07:59 +0000

Refresh of remove-typeptr-and-typehandle-

---

diff --git a/src/global.h b/src/global.h
index 7b28396..4b21931 100644
--- a/src/global.h
+++ b/src/global.h
@@ -118,7 +118,7 @@ typedef struct
 {
   unsigned long f;		/* generic flags */
   unsigned char t[(MAX_LAYER + 1) / 2];	/* thermals */
-} FlagType;
+} FlagType, *FlagTypePtr;
 
 #ifndef __GNUC__
 #define __FUNCTION1(a,b) a ":" #b
@@ -190,7 +190,7 @@ typedef struct			/* holds information about output window */
     fgGC,			/* changed from some routines */
     pmGC;			/* depth 1 pixmap GC to store clip */
 }
-OutputType;
+OutputType, *OutputTypePtr;
 
 /* ----------------------------------------------------------------------
  * layer group. A layer group identifies layers which are always switched
@@ -200,7 +200,7 @@ typedef struct
 {
   Cardinal Number[MAX_LAYER],	/* number of entries per groups */
     Entries[MAX_LAYER][MAX_LAYER + 2];
-} LayerGroupType;
+} LayerGroupType, *LayerGroupTypePtr;
 
 struct BoxType		/* a bounding box */
 {
@@ -212,13 +212,13 @@ typedef struct
 {
   Coord x, y;
   Coord width, height;
-} RectangleType;
+} RectangleType, *RectangleTypePtr;
 
 typedef struct
 {
   char *name;
   char *value;
-} AttributeType;
+} AttributeType, *AttributeTypePtr;
 
 struct AttributeListType
 {
@@ -234,24 +234,24 @@ struct AttributeListType
    based on this. */
 typedef struct {
   ANYOBJECTFIELDS;
-} AnyObjectType;
+} AnyObjectType, *AnyObjectTypePtr;
 
 typedef struct			/* a line/polygon point */
 {
   Coord X, Y, X2, Y2;	/* so Point type can be cast as BoxType */
   long int ID;
-} PointType;
+} PointType, *PointTypePtr;
 
 /* Lines, rats, pads, etc.  */
 typedef struct {
   ANYLINEFIELDS;
-} AnyLineObjectType;
+} AnyLineObjectType, *AnyLineObjectTypePtr;
 
 typedef struct			/* holds information about one line */
 {
   ANYLINEFIELDS;
   char *Number;
-} LineType;
+} LineType, *LineTypePtr;
 
 typedef struct
 {
@@ -261,7 +261,7 @@ typedef struct
   BYTE Direction;
   char *TextString;		/* string */
   void *Element;
-} TextType;
+} TextType, *TextTypePtr;
 
 struct polygon_st			/* holds information about a polygon */
 {
@@ -287,7 +287,7 @@ typedef struct			/* holds information about arcs */
   Coord Width, Height,		/* length of axis */
     X, Y;			/* center coordinates */
   Angle StartAngle, Delta;	/* the two limiting angles in degrees */
-} ArcType;
+} ArcType, *ArcTypePtr;
 
 struct rtree
 {
@@ -313,13 +313,13 @@ typedef struct			/* holds information about one layer */
   AttributeListType Attributes;
   int no_drc; /* whether to ignore the layer when checking the design rules */
 }
-LayerType;
+LayerType, *LayerTypePtr;
 
 typedef struct			/* a rat-line */
 {
   ANYLINEFIELDS;
   Cardinal group1, group2;	/* the layer group each point is on */
-} RatType;
+} RatType, *RatTypePtr;
 
 struct pad_st			/* a SMD pad */
 {
@@ -365,7 +365,7 @@ typedef struct
   GList *Arc;
   BoxType VBox;
   AttributeListType Attributes;
-} ElementType;
+} ElementType, *ElementTypePtr, **ElementTypeHandle;
 
 /* ---------------------------------------------------------------------------
  * symbol and font related stuff
@@ -378,7 +378,7 @@ typedef struct			/* a single symbol */
     LineMax;
   Coord Width, Height,		/* size of cell */
     Delta;			/* distance to next symbol */
-} SymbolType;
+} SymbolType, *SymbolTypePtr;
 
 typedef struct			/* complete set of symbols */
 {
@@ -387,7 +387,7 @@ typedef struct			/* complete set of symbols */
   BoxType DefaultSymbol;	/* the default symbol is a filled box */
   SymbolType Symbol[MAX_FONTPOSITION + 1];
   bool Valid;
-} FontType;
+} FontType, *FontTypePtr;
 
 typedef struct			/* holds all objects */
 {
@@ -403,7 +403,7 @@ typedef struct			/* holds all objects */
   struct PCBType *pcb;
   LayerType Layer[MAX_LAYER + 2];	/* add 2 silkscreen layers */
   int polyClip;
-} DataType;
+} DataType, *DataTypePtr;
 
 typedef struct			/* holds drill information */
 {
@@ -417,14 +417,14 @@ typedef struct			/* holds drill information */
     PinMax;			/* max number of coordinates from malloc() */
   PinType **Pin;		/* coordinates to drill */
   ElementType **Element;	/* a pointer to an array of element pointers */
-} DrillType;
+} DrillType, *DrillTypePtr;
 
 typedef struct			/* holds a range of Drill Infos */
 {
   Cardinal DrillN,		/* number of drill sizes */
     DrillMax;			/* max number from malloc() */
   DrillType *Drill;		/* plated holes */
-} DrillInfoType;
+} DrillInfoType, *DrillInfoTypePtr;
 
 typedef struct
 {
@@ -434,7 +434,7 @@ typedef struct
     Keepaway;			/* min. separation from other nets */
   char *Name;
   int index;
-} RouteStyleType;
+} RouteStyleType, *RouteStyleTypePtr;
 
 /* ---------------------------------------------------------------------------
  * structure used by library routines
@@ -448,7 +448,8 @@ typedef struct
    *Package,			/* package */
    *Value,			/* the value field */
    *Description;		/* some descritional text */
-} LibraryEntryType;
+} LibraryEntryType, *LibraryEntryTypePtr;
+//typedef LibraryEntryType *LibraryEntryTypePtr;
 
 /* If the internal flag is set, the only field that is valid is Name,
    and the struct is allocated with malloc instead of
@@ -466,14 +467,14 @@ typedef struct
   char flag;			/* used by the netlist window to enable/disable nets */
   char internal;		/* if set, this is an internal-only entry, not
 				   part of the global netlist. */
-} LibraryMenuType;
+} LibraryMenuType, *LibraryMenuTypePtr;
 
 typedef struct
 {
   Cardinal MenuN;               /* number of objects */
   Cardinal MenuMax;             /* number of reserved memory locations */
   LibraryMenuType *Menu;      /* the entries */
-} LibraryType;
+} LibraryType, *LibraryTypePtr;
 
 
   /* The PCBType struct holds information about board layout most of which is
@@ -527,7 +528,7 @@ typedef struct PCBType
 
   bool is_footprint;		/* If set, the user has loaded a footprint, not a pcb. */
 }
-PCBType;
+PCBType, *PCBTypePtr;
 
 typedef struct			/* information about the paste buffer */
 {
@@ -535,7 +536,7 @@ typedef struct			/* information about the paste buffer */
   BoxType BoundingBox;
   DataType *Data;		/* data; not all members of PCBType */
   /* are used */
-} BufferType;
+} BufferType, *BufferTypePtr;
 
 /* ---------------------------------------------------------------------------
  * some types for cursor drawing, setting of block and lines
@@ -546,7 +547,7 @@ typedef struct			/* rubberband lines for element moves */
   LayerType *Layer;		/* layer that holds the line */
   LineType *Line;		/* the line itself */
   PointType *MovedPoint;	/* and finally the point */
-} RubberbandType;
+} RubberbandType, *RubberbandTypePtr;
 
 typedef struct			/* current marked line */
 {
@@ -554,7 +555,7 @@ typedef struct			/* current marked line */
     Point2;
   long int State;
   bool draw;
-} AttachedLineType;
+} AttachedLineType, *AttachedLineTypePtr;
 
 typedef struct			/* currently marked block */
 {
@@ -562,7 +563,7 @@ typedef struct			/* currently marked block */
     Point2;
   long int State;
   bool otherway;
-} AttachedBoxType;
+} AttachedBoxType, *AttachedBoxTypePtr;
 
 typedef struct			/* currently attached object */
 {
@@ -576,7 +577,7 @@ typedef struct			/* currently attached object */
   Cardinal RubberbandN,		/* number of lines in array */
     RubberbandMax;
   RubberbandType *Rubberband;
-} AttachedObjectType;
+} AttachedObjectType, *AttachedObjectTypePtr;
 
 enum crosshair_shape
 {
@@ -598,13 +599,13 @@ typedef struct			/* holds cursor information */
   PolygonType AttachedPolygon;
   AttachedObjectType AttachedObject;	/* data of attached objects */
   enum crosshair_shape shape; 	/* shape of crosshair */
-} CrosshairType;
+} CrosshairType, *CrosshairTypePtr;
 
 typedef struct
 {
   bool status;
   Coord X, Y;
-} MarkType;
+} MarkType, *MarkTypePtr;
 
 /* ---------------------------------------------------------------------------
  * our resources
@@ -695,7 +696,7 @@ typedef struct			/* some resources... */
     AutoPlace;			/* flag which says we should force placement of the
 				   windows on startup */
 }
-SettingType;
+SettingType, *SettingTypePtr;
 
 /* ----------------------------------------------------------------------
  * pointer to low-level copy, move and rotate functions
@@ -714,7 +715,7 @@ typedef struct
   void *(*Point) (LayerType *, PolygonType *, PointType *);
   void *(*Arc) (LayerType *, ArcType *);
   void *(*Rat) (RatType *);
-} ObjectFunctionType;
+} ObjectFunctionType, *ObjectFunctionTypePtr;
 
 /* ---------------------------------------------------------------------------
  * structure used by device drivers
@@ -727,7 +728,7 @@ typedef struct			/* holds a connection */
   void *ptr1, *ptr2;		/* the object of the connection */
   Cardinal group;		/* the layer group of the connection */
   LibraryMenuType *menu;	/* the netmenu this *SHOULD* belong too */
-} ConnectionType;
+} ConnectionType, *ConnectionTypePtr;
 
 typedef struct			/* holds a net of connections */
 {
@@ -735,28 +736,28 @@ typedef struct			/* holds a net of connections */
     ConnectionMax;		/* max connections from malloc */
   ConnectionType *Connection;
   RouteStyleType *Style;
-} NetType;
+} NetType, *NetTypePtr;
 
 typedef struct			/* holds a list of nets */
 {
   Cardinal NetN,		/* the number of subnets contained */
     NetMax;			/* max subnets from malloc */
   NetType *Net;
-} NetListType;
+} NetListType, *NetListTypePtr;
 
 typedef struct			/* holds a list of net lists */
 {
   Cardinal NetListN,		/* the number of net lists contained */
     NetListMax;			/* max net lists from malloc */
   NetListType *NetList;
-} NetListListType;
+} NetListListType, *NetListListTypePtr;
 
 typedef struct			/* holds a generic list of pointers */
 {
   Cardinal PtrN,		/* the number of pointers contained */
     PtrMax;			/* max subnets from malloc */
   void **Ptr;
-} PointerListType;
+} PointerListType, *PointerListTypePtr;
 
 typedef struct
 {
@@ -764,7 +765,7 @@ typedef struct
     BoxMax;			/* max boxes from malloc */
   BoxType *Box;
 
-} BoxListType;
+} BoxListType, *BoxListTypePtr;
 
 struct drc_violation_st
 {
diff --git a/src/mymem.h b/src/mymem.h
index 46061a3..71c0a84 100644
--- a/src/mymem.h
+++ b/src/mymem.h
@@ -62,7 +62,7 @@ typedef struct
 {
   size_t MaxLength;
   char *Data;
-} DynamicStringType;
+} DynamicStringType, *DynamicStringTypePtr;
 
 RubberbandType * GetRubberbandMemory (void);
 PinType * GetPinMemory (ElementType *);
